src/features/annotation-thread/AnnotationThreadContent.js

```diff
@@ -0,0 +1,192 @@
+    };
+
+    const handleAnnotationDelete = ({ id, permissions }: { id: string, permissions: AnnotationPermission }) => {
+        api.getAnnotationsAPI(false).deleteAnnotation(fileId, id, permissions, noop, annotationErrorCallback);
+    };
+
+    const handleAnnotationEdit = (id: string, text: string, permissions: AnnotationPermission) => {
+        api.getAnnotationsAPI(false).updateAnnotation(
+            fileId,
+            id,
+            permissions,
+            { message: text },
+            annotationSuccessCallback,
+            annotationErrorCallback,
+        );
+    };
+
+    const handleAnnotationResolve = (id: string, status: FeedItemStatus, permissions: AnnotationPermission) => {
+        api.getAnnotationsAPI(false).updateAnnotation(
+            fileId,
+            id,
+            permissions,
+            { status },
+            annotationSuccessCallback,
+            annotationErrorCallback,
+        );
+    };
+
+    /**
+     * File @mention contacts fetch success callback
+     *
+     * @private
+     * @param {BoxItemCollection} collaborators - Collaborators response data
+     * @return {void}
+     */
+    const getMentionContactsSuccessCallback = (collaborators: { entries: SelectorItems<> }): void => {
+        setMentionSelectorContacts(collaborators.entries);
+    };
+
+    /**
+     * Fetches file collaborators
+     *
+     * @param {Function} successCallback - the success callback
+     * @param {Function} onErrorCallback - the error callback
+     * @param {string} searchStr - the search string
+     * @param {Object} [options]
+     * @param {boolean} [options.includeGroups] - return groups as well as users
+     * @return {void}
+     */
+    const getCollaborators = (
+        successCallback: Collaborators => void,
+        onErrorCallback: ElementsErrorCallback,
+        searchStr: string,
+        { includeGroups = false }: { includeGroups: boolean } = {},
+    ): void => {
+        // Do not fetch without filter
+        if (!searchStr || searchStr.trim() === '') {
+            return;
+        }
+
+        api.getFileCollaboratorsAPI(true).getFileCollaborators(fileId, successCallback, onErrorCallback, {
+            filter_term: searchStr,
+            include_groups: includeGroups,
+            include_uploader_collabs: false,
+        });
+    };
+
+    /**
+     * Fetches file @mention's
+     *
+     * @private
+     * @param {string} searchStr - Search string to filter file collaborators by
+     * @return {void}
+     */
+    const getMentionWithQuery = debounce(
+        (searchStr: string) => getCollaborators(getMentionContactsSuccessCallback, errorCallback, searchStr),
+        DEFAULT_COLLAB_DEBOUNCE,
+    );
+
+    const getAvatarUrl = async (userId: string): Promise<?string> => {
+        return api.getUsersAPI(false).getAvatarUrlWithAccessToken(userId, fileId);
+    };
+
+    const getUserProfileUrl = (userID: string) => Promise.resolve(`/profile/${userID}`);
```

Comment:
`ActivitySidebar` seems to have similar functionality, but calls
```
getAvatarUrl = async (userId: string): Promise<?string> => {
        const { file, api } = this.props;

        return api.getUsersAPI(false).getAvatarUrlWithAccessToken(userId, file.id);
    };
```
Would that work here?

Reply:
this one is for getting Avatar. `getUserProfileUrl` should get user profile and actually the function should be exposed as external prop in `AnnotationThread`, I fixed it 👍